Kubernetes/legos/k8s_get_memory_utilization_of_services/k8s_get_memory_utilization_of_services.py

```diff
@@ -131,38 +124,72 @@ def k8s_get_memory_utilization_of_services(handle, namespace: str = "", threshol
                 if not svc_pods:
                     # No pods attached to the given service
                     continue
+
+                # For each pod, fetch containers and their memory usage
                 for svc_pod in svc_pods.split():
-                    if svc_pod in pod_mem_util_dict.keys():
-                        pods_to_check[svc_pod] = pod_mem_util_dict[svc_pod]
-
-        if not pods_to_check:
-            pods_to_check = pod_mem_util_dict
-
-        for pod, mem_usage in pods_to_check.items():
-            # Lets get memory requests for all containers not just the first container
-            kubectl_command = f"kubectl get pod {pod} -n {namespace} -o=jsonpath='{{.spec.containers[*].resources.requests.memory}}'"
-            response = handle.run_native_cmd(kubectl_command)
-            mem_request = response.stdout.strip()
-            _mem_request = 0
-            for _mr in mem_request.split():
-                # Lets calculate the total memory request for all containers in the POD
-                _mem_request += convert_memory_to_bytes(_mr)
-            mem_request = _mem_request
-            mem_usage = convert_memory_to_bytes(mem_usage)
-
-            if not mem_request:
-                # Memory limit is not set, dont calculate utilization
-                continue
-            utilization = (mem_usage / mem_request) * 100
-            utilization = round(utilization, 2)
-            if utilization > threshold:
-                exceeding_services.append({
-                    "pod": pod,
-                    "namespace": namespace,
-                    "utilization_percentage": utilization,
-                    "memory_request_bytes": mem_request,
-                    "memory_usage_bytes": mem_usage,
-                })
+                    top_pod_containers_cmd = f"kubectl top pod {svc_pod} -n {namespace} --containers --no-headers"
+                    response = handle.run_native_cmd(top_pod_containers_cmd)
+                    top_pod_containers_output = response.stdout.strip()
+                    if not top_pod_containers_output:
+                        continue
+
+                    for line in top_pod_containers_output.split('\n'):
+                        parts = line.split()
+                        if len(parts) >= 3:  # Ensure line has enough parts to parse
+                            container_name = parts[1]
+                            mem_usage = parts[-1]
```

Comment:
What is the case of `else`?
Do we need to handle it?

Reply:
we may have to print an error and continue in that case, can't stop the execution flow.

Reply:
Done